db/upgradelib.php

```diff
@@ -69,25 +69,32 @@ function surveypro_delete_supposed_blank_answers() {
     // Parent items without a saved answer must be considered as parent NOT allowing children.
     // Delete all the answers to children items that have parent without saved answers.
     // Alias: if the parent was not allowed in the userform, its children will not be allowed even more.
-    $sql = 'SELECT id as childid, parentid, surveyproid
+    $sql = 'SELECT parentid, MAX(surveyproid) as surveyproid
             FROM {surveypro_item}
             WHERE type = :type
                 AND parentid <> :parentid
             GROUP BY parentid
-            ORDER BY surveyproid, sortindex';
+            ORDER BY 2';
```

Comment:
While this works ok under majority of databases... I'd recommend to order by "MAX(surveyproid)" (or its alias, although in this case it can be missleading, because the alias matches a field name) instead. Mainly because that's the way we do it in Moodle (you won't find many "ORDER by position" occurrences.